### PR TITLE
photos: improve user ID generation

### PR DESCRIPTION
**Pull request by petermattis**
Hash user IDs after generation to avoid having a single range contain
all of the "hot" users.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/examples-go/92)

### CODE REVIEW

File: photos/user.go [hunk 1]
```diff
@@ -138,8 +141,21 @@ func startStats(stopper *stop.Stopper) {
 // startUser simulates a stream of user events until the stopper
 // indicates it's time to exit.
 func startUser(ctx Context, stopper *stop.Stopper) {
+	h := fnv.New32()
+	var buf [8]byte
+
+	randomUser := func() int {
+		// User an exponential distribution to skew the user ID generation, but
```
Comment (bdarnell) [inline photos/user.go]: s/User/use/

File: photos/user.go [hunk 2]
```diff
@@ -138,8 +141,21 @@ func startStats(stopper *stop.Stopper) {
...
+		// hash the randomly generated value so that the "hot" users are spread
+		// throughout the user ID key space (and thus not all on 1 range).
+		binary.BigEndian.PutUint64(buf[:8], math.Float64bits(rand.ExpFloat64()/rate))
+		h.Reset()
+		h.Write(buf[:8])
+		return 1 + int(h.Sum32())
```
Comment (bdarnell) [inline photos/user.go]: Can probably remove the `1+`.

File: photos/user.go (hunk 1 above)
Comment (petermattis) [inline photos/user.go]: Done.

File: photos/user.go (hunk 2 above)
Comment (petermattis) [inline photos/user.go]: Yeah, I'm not sure what that was doing in the first place. Removed.

Comment (bdarnell) [inline photos/user.go]: I bet it was trying to make sure it never generated a user ID of zero (although that doesn't really matter).

**Pull request merged**